Cowles Foundation.js

```diff
@@ -0,0 +1,322 @@
+		item.pages = ZU.xpathText(doc, "//strong[contains(., 'CFP page numbers')]/following-sibling::text()").split(',')[0];
+		author = doc.querySelectorAll("div.comma span.comma a");
+		for (let auth of author) item.creators.push(ZU.cleanAuthor(auth.textContent, "author", false));
+		item.url = ZU.xpathText(doc, "//strong[contains(., 'CFP Paper Title')]/following-sibling::a/@href");
+		item.libraryCatalog = "Cowles Foundation";
+		try {
+			let seeAlso = "See also: " + ZU.xpathText(doc, "//strong[contains(., 'See CFDP')]/following-sibling::span//text()");
+			if (seeAlso != "See also: null") item.notes.push({ note:seeAlso });
+		}
+		catch (err) {}
+	}
+	else if (url.includes('/cfdp/')) {
+		item = new Zotero.Item("report");
+		item.title = text(doc, 'h3 a');
+		item.reportType = "Cowles Foundation Discussion Paper";
+		item.reportNumber = text(doc, "#page-title").match(/\d+/)[0];
+		author = doc.querySelectorAll("div.comma span.comma a");
+		for (let auth of author) item.creators.push(ZU.cleanAuthor(auth.textContent, "author", false));
+
+		item.libraryCatalog = "Cowles Foundation";
+		item.date = ZU.strToISO(ZU.xpathText(doc, "//strong[contains(., 'Publication Date')]/following-sibling::span//text()"));
+		item.pages = ZU.xpathText(doc, "//strong[contains(., 'Pages')]/following-sibling::text()");
+		try {
+			item.abstractNote = ZU.xpathText(doc, "//p[contains(., 'Abstract')]/following-sibling::p/text()");
+		}
+		catch (err) {}
+		pdfurl = attr(doc, 'h3 a', 'href');
+		item.attachments.push({
+			title: item.title,
+			mimeType: "application/pdf",
+			url: pdfurl
+		});
+	}
+	else if (url.includes('/cfm-')) {
+		item = new Zotero.Item("book");
+		item.title = text(doc, 'a[href*="/pub/"]');
+		item.series = "Cowles Monograph";
+		item.seriesNumber = text(doc, "#page-title").match(/\d+/)[0];
+		item.libraryCatalog = "Cowles Foundation";
+		
+		var creatorType = "author";
+		var author = doc.querySelector(".field-name-field-paper-title p a")
+		.previousSibling
+		.textContent
+		.trim()
+		.split(",");
+		if (author[1] == " ed." || author[1] == " eds.") creatorType = "editor";
+		else creatorType = "author";
+		
+		if (author[0].includes(" and ")) {
+			var authors = author[0].split(" and ");
+			for (let auth of authors){
+				item.creators.push(ZU.cleanAuthor(auth, creatorType, false));
+			}
+		}
+		else if (author[0].includes(" & ")) item.creators.push(ZU.cleanAuthor(author[0].split(" & ")[0], creatorType, false));
+		else item.creators.push(ZU.cleanAuthor(author[0], creatorType, false));
+		
+		try {
+			var editor = doc.querySelector(".field-name-field-paper-title p a").nextSibling.textContent.trim().split(", ");
+			if (editor[1].includes("ed.")) {
+				item.edition = editor[1];
+				item.publisher = editor[2];
+			}
+			else item.publisher = editor[1];
+			item.date = ZU.strToISO(editor[editor.length - 1].split(" ")[0]);
+		}
+		catch (err) {}
+		
+		pdfurl = attr(doc, 'a[href*="/pub/"]', 'href');
+		item.attachments.push({
```

Comment:
Could you pull the `item.attachments.push` calls out of the various `if` branches? You're setting one `pdfurl` variable anyway.